configure.ac

```diff
@@ -0,0 +1,149 @@
+AC_INIT([mib2h5], [0.0.1], [https://github.com/ePSIC-DLS/mib2h5/issues], [mib2h5], [https://github.com/ePSIC-DLS/mib2h5])
+AM_INIT_AUTOMAKE([-Wall foreign -Werror tar-pax dist-xz])
+AC_PROG_CC
+m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
+AC_CONFIG_HEADERS([config.h])
+
+# --- Build Type Options --
+AC_ARG_ENABLE([debug],
+  AS_HELP_STRING([--enable-debug], [Enable debug build]),
+  [debug_build=$enableval], [debug_build=no])
+
+AC_ARG_ENABLE([asan],
+  AS_HELP_STRING([--enable-asan], [Enable AddressSanitizer build]),
+  [asan_build=$enableval], [asan_build=no])
+
+# --- Check for HDF5 ---
+
+AC_ARG_WITH([hdf5],
+  [AS_HELP_STRING([--with-hdf5=PATH], [Path to HDF5 installation])],
+  [HDFDIR="$withval"],
+  [])
+
+AC_MSG_NOTICE([Check the path of --with-hdf5])
+AS_IF([test -n "$HDFDIR"], [
+  # User gave --with-hdf5 — use it directly
+  CPPFLAGS="$CPPFLAGS -I$HDFDIR/include"
+  LDFLAGS="$LDFLAGS -L$HDFDIR/lib -Wl,-rpath=$HDFDIR/lib"
+
+  AC_CHECK_HEADER([hdf5.h], [], [AC_MSG_ERROR([Missing hdf5.h in $HDFDIR/include])])
+  AC_CHECK_LIB([hdf5], [H5open], [], [AC_MSG_ERROR([Missing libhdf5 in $HDFDIR/lib])])
+  AC_DEFINE([HAVE_HDF5], [1], [Define if HDF5 is available])
+],
+[
+  # No --with-hdf5: try different common path
+  AC_MSG_NOTICE([Check the path of $HDF5_ROOT])
+  AS_IF([test -n "$HDF5_ROOT"], [
+    CPPFLAGS="$CPPFLAGS -I$HDF5_ROOT/include"
+    LDFLAGS="$LDFLAGS -L$HDF5_ROOT/lib -Wl,-rpath=$HDF5_ROOT/lib"
+
+    AC_CHECK_HEADER([hdf5.h], [], [AC_MSG_ERROR([Missing hdf5.h in $HDF5_ROOT/include])])
+    AC_CHECK_LIB([hdf5], [H5open], [], [AC_MSG_ERROR([Missing libhdf5 in $HDF5_ROOT/lib])])
+    AC_DEFINE([HAVE_HDF5], [1], [Define if HDF5 is available])
+  ],
+  [
+    AC_MSG_NOTICE([Check the path of $HDF5_HOME])
+    AS_IF([test -n "$HDF5_HOME"], [
+      CPPFLAGS="$CPPFLAGS -I$HDF5_HOME/include"
+      LDFLAGS="$LDFLAGS -L$HDF5_HOME/lib -Wl,-rpath=$HDF5_HOME/lib"
+
+      AC_CHECK_HEADER([hdf5.h], [], [AC_MSG_ERROR([Missing hdf5.h in $HDF5_HOME/include])])
+      AC_CHECK_LIB([hdf5], [H5open], [], [AC_MSG_ERROR([Missing libhdf5 in $HDF5_HOME/lib])])
+      AC_DEFINE([HAVE_HDF5], [1], [Define if HDF5 is available])
+    ],
+    [
+      AC_MSG_NOTICE([Check the path of $HDF5_DIR])
+      AS_IF([test -n "$HDF5_DIR"], [
+        CPPFLAGS="$CPPFLAGS -I$HDF5_DIR/include"
+        LDFLAGS="$LDFLAGS -L$HDF5_DIR/lib -Wl,-rpath=$HDF5_DIR/lib"
+
+        AC_CHECK_HEADER([hdf5.h], [], [AC_MSG_ERROR([Missing hdf5.h in $HDF5_DIR/include])])
+        AC_CHECK_LIB([hdf5], [H5open], [], [AC_MSG_ERROR([Missing libhdf5 in $HDF5_DIR/lib])])
+        AC_DEFINE([HAVE_HDF5], [1], [Define if HDF5 is available])
+      ],
```

Comment:
You can avoid the duplicated codes by assigning the corresponding environment variable to `HDFDIR` and perform only one go of `CPPFLAGS`/`LDFLAGS` and header/lib check.